DIP upload fixes, refs #13067 and #13068

jraddaoui opened the pull request:
**Get namespaces from METS file when possible:**

Use `getDocNamespaces` to get the namespaces declared in the METS file
dynamically from the document. To keep backwards compatibility, add the
old URI for main namespaces if they're not found in the document.

Use a key/name mapping to add those namespaces in XPath queries without
modifying the existing ones.

**Allow different namespaces in DC metadata:**

Allow no namespace and `dc` or `dcterms` namespaces.

**Fix main DMD section finding:**

Use always the physical (default) structMap.

**Fix all DMD section finding:**

The DMDID attribute can contain one or more DMD section ids
(e.g.: DMDID="dmdSec_2 dmdSec_3"). When multiple DMD sections
are associated with the same file/dir we'll try to return the
latest one created.

replaceafill reviewed (approved):
The code looks great to me @jraddaoui! :+1: 

I tested it with a `DemoTransferCSV` transfer and each object appears in AtoM with its correct metadata. I also arranged a SIP from the backlog with multiple directories, files and levels of description and they showed correctly too. However, I couldn't figure out a way to test a DMDID with multiple values. Can you recommend a way to do that?

Also, out of curiosity, in https://projects.artefactual.com/issues/13067#note-10 you mention:

> it was not creating any archival description before the fix. It should create descriptions now

Could you clarify how can that be tested? Thanks!

jraddaoui committed:
DIP upload fixes, refs #13067 and #13068

Squashed commit of the following:

commit 4764110f9426d4fc288b2e64bd185a7fb3d0c053
Author: José Raddaoui Marín <raddaouimarin@gmail.com>
Date:   Mon Aug 12 15:26:15 2019 +0200

    Fix all DMD section finding

    The DMDID attribute can contain one or more DMD section ids
    (e.g.: DMDID="dmdSec_2 dmdSec_3"). When multiple DMD sections
    are associated with the same file/dir we'll try to return the
    latest one created.

commit 253d196f82a704ae76366e746d22109979eff98e
Author: José Raddaoui Marín <raddaouimarin@gmail.com>
Date:   Fri Aug 9 22:24:40 2019 +0200

    Fix main DMD section finding

    Use always the physical (default) structMap.

commit c8dda44b29bbb56a9c8c00c00ef148af177c3a40
Author: José Raddaoui Marín <raddaouimarin@gmail.com>
Date:   Fri Aug 9 20:47:40 2019 +0200

    Allow different namespaces in DC metadata

    Allow no namespace and `dc` or `dcterms` namespaces.

commit 06e7f08d64e51607c9639fd4e449976bbe8fbb80
Author: José Raddaoui Marín <raddaouimarin@gmail.com>
Date:   Fri Aug 9 20:45:04 2019 +0200

    Get namespaces from METS file when possible

    Use `getDocNamespaces` to get the namespaces declared in the
    METS file dynamically from the document. To keep backwards
    compatibility, add the old URI for main namespaces if they're
    not found in the document.

    Use a key/name mapping to add those namespaces in XPath queries
    without modifying the existing ones.

jraddaoui commented:
Thanks @replaceafill!

I appreciate that you gave it a try too. We don't normally do that as this will be tested by the analysts before the release (I did it yesterday on your AM PR because I needed those changes to finalize this work :).

To answer your questions ...

> I couldn't figure out a way to test a DMDID with multiple values. Can you recommend a way to do that?

That requires a partial re-ingest in AM, entering DC metadata on both ingest process.

> Could you clarify how can that be tested?

You already did, no description was being created when the upload was performed from a SIP created in the arrange panel.
